### PR TITLE
PXC 8.0.27 refresh - Q4 2021

### PR DESCRIPTION
**Pull request by kamil-holubicki**
Server part updated to PS 8.0.27
Galera provider updated to 26.4.10
wsrep-lib updated
wsrep-API updated

### CODE REVIEW

File: unittest/gunit/group_replication/group_replication_gcs_mysql_network_provider-t.cc [hunk 1]
```diff
@@ -427,6 +427,13 @@ TEST_F(MySQLNetworkProviderTest, NewServerConnectionTest) {
 
   constexpr int socket_to_use = 42;
 
+#ifdef WITH_WSREP
+  wsrep::gtid gtid;
+  Wsrep_server_state::init_once(std::string(), std::string(), std::string(),
+                            std::string(), gtid,
```
Comment (venkatesh-prasad-v) [inline unittest/gunit/group_replication/group_replication_gcs_mysql_network_provider-t.cc]: Please run clang-format-10 on all files touched by this commit (e867c567c7da6edf95e019bc729ad6421c023b94)

File: sql/wsrep_mysqld.cc [hunk 2]
```diff
@@ -313,19 +313,19 @@ PSI_thread_key key_THREAD_galera_receiver;
...
      PSI_DOCUMENT_ME},
-    {&key_THREAD_galera_ist_async_sender, "THREAD_galera_ist_async_sender", 0,
+    {&key_THREAD_galera_ist_async_sender, "THREAD_galera_ist_async_sender", "ist_sender",0,
      0, PSI_DOCUMENT_ME},
-    {&key_THREAD_galera_writeset_checksum, "THREAD_galera_writeset_checksum", 0,
+    {&key_THREAD_galera_writeset_checksum, "THREAD_galera_writeset_checksum", "wset_check",0,
      0, PSI_DOCUMENT_ME},
-    {&key_THREAD_galera_gcache_removefile, "THREAD_galera_gcache_removefile", 0,
+    {&key_THREAD_galera_gcache_removefile, "THREAD_galera_gcache_removefile", "gcache_rm",0,
      0, PSI_DOCUMENT_ME},
```
Comment (venkatesh-prasad-v) [inline sql/wsrep_mysqld.cc]: Can we have the new names given to galera threads start with `galera_`?

File: storage/innobase/handler/ha_innodb.cc [hunk 3]
```diff
@@ -8865,7 +8866,7 @@ uint wsrep_store_key_val_for_row(THD *thd, TABLE *table, uint keynr, char *buff,
 
       blob_data = row_mysql_read_blob_ref(
           &blob_len, (byte *)(record + (ulint)get_field_offset(table, field)),
-          (ulint)field->pack_length(), false, 0, 0, prebuilt);
+          (ulint)field->pack_length(), false, 0, 0, &prebuilt->compress_heap);
```
Comment (venkatesh-prasad-v) [inline storage/innobase/handler/ha_innodb.cc]: Why this change?

File: storage/innobase/handler/ha_innodb.cc [hunk 4]
```diff
@@ -10639,7 +10640,7 @@ static int wsrep_calc_row_hash(byte *digest, const uchar *row, TABLE *table,
 
     switch (col_type) {
       case DATA_BLOB:
-        ptr = row_mysql_read_blob_ref(&len, ptr, len, false, 0, 0, prebuilt);
+        ptr = row_mysql_read_blob_ref(&len, ptr, len, false, 0, 0, &prebuilt->compress_heap);
```
Comment (venkatesh-prasad-v) [inline storage/innobase/handler/ha_innodb.cc]: Why this change?

File: sql/mysqld.cc [hunk 5]
```diff
@@ -13200,12 +13201,12 @@ PSI_FLAG_USER | PSI_FLAG_NO_SEQNUM, 0, PSI_DOCUMENT_ME},
...
-  { &key_THREAD_wsrep_sst_joiner, "THREAD_wsrep_sst_joiner", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME},
-  { &key_THREAD_wsrep_sst_donor, "THREAD_wsrep_sst_donor", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME},
-  { &key_THREAD_wsrep_sst_logger, "THREAD_wsrep_sst_logger", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME},
-  { &key_THREAD_wsrep_applier, "THREAD_wsrep_applier", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME},
-  { &key_THREAD_wsrep_rollbacker, "THREAD_wsrep_rollbacker", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME},
-  { &key_THREAD_wsrep_post_rollbacker, "THREAD_wsrep_post_rollbacker", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME}
+  { &key_THREAD_wsrep_sst_joiner, "THREAD_wsrep_sst_joiner", "sst_joiner", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME},
+  { &key_THREAD_wsrep_sst_donor, "THREAD_wsrep_sst_donor", "sst_donor", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME},
+  { &key_THREAD_wsrep_sst_logger, "THREAD_wsrep_sst_logger", "sst_logger", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME},
+  { &key_THREAD_wsrep_applier, "THREAD_wsrep_applier", "applier", PSI_FLAG_SINGLETON, 0, PSI_DOCUMENT_ME},
```
Comment (venkatesh-prasad-v) [inline sql/mysqld.cc]: Can we name it as `wsrep_applier` instead if just `applier`?

File: mysql-test/mysql-test-run.pl [hunk 6]
```diff
@@ -300,18 +300,7 @@ sub set_term_args {
   percona-pam-for-mysql
   data_masking
   procfs
-  rocksdb
-  rocksdb_rpl
-  rocksdb_sys_vars
-  rocksdb_stress
   rpl_encryption
-  tokudb
```
Comment (venkatesh-prasad-v) [inline mysql-test/mysql-test-run.pl:307]: Is it a better idea to remove the test files as well?

**Review by venkatesh-prasad-v — CHANGES_REQUESTED** [review]
Few comments

1. Please update the `XB_8x_REQUIRED_VERSION` in `scripts/wsrep_sst_xtrabackup-v2.sh` to 8.0.27.
2. Please update the xtrabackup tag used in `cmake/dev_helper.cmake` file.

File: sql/wsrep_mysqld.cc (hunk 2 above)
Comment (kamil-holubicki) [reply sql/wsrep_mysqld.cc]: Unfortunately we can't. There is limitation in register_thread_class()
``` assert(strlen(info->m_os_name) < PFS_MAX_OS_NAME_LENGTH - 3); ```
and
```#define PFS_MAX_OS_NAME_LENGTH 16```
so we've got only 12 characters

File: unittest/gunit/group_replication/group_replication_gcs_mysql_network_provider-t.cc (hunk 1 above)
Comment (kamil-holubicki) [reply unittest/gunit/group_replication/group_replication_gcs_mysql_network_provider-t.cc]: done

File: storage/innobase/handler/ha_innodb.cc (hunk 4 above)
Comment (kamil-holubicki) [reply storage/innobase/handler/ha_innodb.cc]: ```row_mysql_read_blob_ref()``` declaration has changed

File: storage/innobase/handler/ha_innodb.cc (hunk 3 above)
Comment (kamil-holubicki) [reply storage/innobase/handler/ha_innodb.cc]: ```row_mysql_read_blob_ref()``` declaration has changed

File: sql/wsrep_mysqld.cc (hunk 2 above)
Comment (venkatesh-prasad-v) [reply sql/wsrep_mysqld.cc]: Ok. Then let us go with the current names only.

File: sql/mysqld.cc (hunk 5 above)
Comment (kamil-holubicki) [reply sql/mysqld.cc]: likewise. Limitation.

File: mysql-test/mysql-test-run.pl (hunk 6 above)
Comment (kamil-holubicki) [reply mysql-test/mysql-test-run.pl:307]: Removed all TokuDB related MTR suites.

**Review by venkatesh-prasad-v — APPROVED** [review]
Please revert the changes in `.gitmodules` before merging.

**Comment by kamil-holubicki** [comment]
https://pxc.cd.percona.com/view/PXC%208.0/job/pxc-8.0-param/434/

**Pull request merged**